src/openrct2/world/map.c

```diff
@@ -3909,8 +3909,8 @@ bool sign_set_colour(sint32 x, sint32 y, sint32 z, sint32 direction, sint32 sequ
         z = (z0 + tile->z_offset) / 8;
         mapElement = map_get_large_scenery_segment(x, y, z, direction, sequence);
         if (mapElement != NULL) {
-            mapElement->properties.scenerymultiple.colour[0] &= 0xE0;
-            mapElement->properties.scenerymultiple.colour[1] &= 0xE0;
+            mapElement->properties.scenerymultiple.colour[0] = SCENERY_COLOUR_PROPERTIES(mapElement->properties.scenerymultiple.colour[0]);
```

Comment:
This does not look correct. I think there should be a function to set primary and secondary colour for large scenery map elements.

Reply:
Do you think `mapElement->properties.scenerymultiple.colour[0] &= SCENERY_COLOUR_PROPERTIES_MASK;` would be better?

Reply:
No, a helper function like scenery_small_get_primary_colour().
But I added it, since I was working in tile elements anyway.